tests/acceptance/serverd-multi-versions.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+PWD=`pwd`
+logdir=$PWD/serverd-multi-versions-logs
+statsdir=$PWD/serverd-multi-versions-stats
+
+mkdir -p hub
+mkdir -p client
+mkdir -p $logdir
+mkdir -p $statsdir
+cat >server.sh <<EOF
+#!/bin/sh
+
+LD_LIBRARY_PATH=$PWD/hub/var/cfengine/lib CFENGINE_TEST_OVERRIDE_EXTENSION_LIBRARY_DIR=$PWD/hub/var/cfengine/lib CFENGINE_TEST_OVERRIDE_WORKDIR=$PWD/work_here $PWD/hub/var/cfengine/bin/cf-serverd "\$@"
+EOF
+cat >agent.sh <<EOF
+#!/bin/sh
+LD_LIBRARY_PATH=$PWD/agent/var/cfengine/lib CFENGINE_TEST_OVERRIDE_EXTENSION_LIBRARY_DIR=$PWD/agent/var/cfengine/lib CFENGINE_TEST_OVERRIDE_WORKDIR=$PWD/work_here $PWD/agent/var/cfengine/bin/cf-agent "\$@"
+EOF
+chmod a+x *.sh
+sed -i "s%\$(sys.cf_agent)%$PWD/agent.sh%g;s%\$(sys.cf_serverd)%$PWD/server.sh%g" tests/acceptance/run_with_server.cf.sub
+chmod -R 700 tests
+
+mkdir -p work_stub/inputs/
+cp -a $PWD/tests/acceptance work_stub/inputs/
+rm -rf work_stub/inputs/acceptance/0*
+mkdir -p work_stub/bin/
+touch work_stub/bin/cf-promises
+chmod a+x work_stub/bin/cf-promises
+
+sudo hostname localhost
+# Ensure reverse hostname resolution is correct and 127.0.0.1 is always 'localhost'.
+# There's no nice shell command to test it but this one:
+# python -c 'import socket;print socket.gethostbyaddr("127.0.0.1")'
+sudo sed -i -e '1s/^/127.0.0.1 localhost localhost.localdomian\n/' /etc/hosts
+
+# downloads $3 (if provided), saves it as $2 and extracts it to $1
+# if $2 not provided - installs currently built version
+function install(){
+    test -z "$1" && return 1
+    rm -rf $PWD/$1
```

Comment:
Why do you prepend `$PWD` to the arguments? Isn't `rm -rf "$1"` working?

EDIT: I'm a bit confused by the $PWDs all over. Why?

Reply:
Also use double quotes around the expansions with `rm -rf` for safety.

Reply:
right.